src/app/proposals/components/prepare-proposal-dropdown/prepare-proposal-dropdown.tsx

```diff
@@ -0,0 +1,88 @@
+import { HTMLAttributes, useState, useRef } from 'react'
+import { motion, AnimatePresence } from 'motion/react'
+import { ChevronDown, X, ChevronRight } from 'lucide-react'
+import { cn } from '@/lib/utils'
+import { HeaderTitle, Typography } from '@/components/Typography'
+import { useClickOutside } from '@/shared/hooks/useClickOutside'
+import { dropdownItemsData } from './data'
+
+/**
+ * "Prepare Your Proposal" custom dropdown menu with smooth animation
+ */
+export function PrepareProposalDropdown({ className, ...props }: HTMLAttributes<HTMLDivElement>) {
+  const [isOpen, setIsOpen] = useState(false)
+  const toggle = () => setIsOpen(!isOpen)
+  const close = () => setIsOpen(false)
+
+  // Closes the dropdown when clicking outside
+  const dropdownRef = useRef<HTMLDivElement>(null)
+  useClickOutside([dropdownRef], close)
+
+  const openLink = (link: string) => {
+    close()
+    // Timeout ensures the menu closes before opening the link to prevent UI glitches.
+    setTimeout(() => {
+      window.open(link, '_blank', 'noopener,noreferrer')
+    }, 200)
+  }
+
+  return (
+    <div {...props} ref={dropdownRef} className={cn(className, 'relative w-full max-w-[380px]')}>
+      <button
+        onClick={toggle}
+        className={cn(
+          'px-5 w-full h-12 flex items-center justify-between gap-2 bg-[#E4E1DA] rounded-t-[4px] focus:outline-none',
+          !isOpen && 'rounded-b-[4px]',
+        )}
+        data-testid="PrepareYourProposalDropdown"
+      >
+        <HeaderTitle className="text-lg leading-none text-black whitespace-nowrap">
+          Prepare your proposal
+        </HeaderTitle>
+        {isOpen ? <X className="text-black cursor-pointer" /> : <ChevronDown className="text-black" />}
+      </button>
+
+      <AnimatePresence>
+        {isOpen && (
+          <motion.div
+            initial={{ opacity: 0, y: -10 }}
+            animate={{ opacity: 1, y: 0 }}
+            exit={{ opacity: 0, y: -10 }}
+            transition={{ duration: 0.2 }}
+            className="absolute left-0 top-full w-full bg-[#E4E1DA] rounded-b-[4px] overflow-hidden z-10 shadow-lg"
+          >
+            <Typography className="px-5 max-w-[330px] text-[#666057] text-[15px] leading-snug">
+              If these steps are not completed, your proposal is unlikely to reach quorum and succeed in a
+              vote.
+            </Typography>
+
+            <div className="py-3 flex flex-col">
+              {dropdownItemsData.map(({ Icon, text, title, id, linkUrl }) => (
+                <div
+                  key={id}
+                  onClick={() => openLink(linkUrl)}
```

Comment:
It's good practice to avoid declaring functions in the props, and instead have them created before the return; because this will be recreated on each render, meaning that the component will trigger a re-render every time. My advise would be to move this outside the return - and also outside of React if possible - and instead access the linkUrl through other means such as name, data attributes, etc. This comment is not necessary for this PR to pass.

Reply:
I totally agree, and I always try moving all possible functions outside react components. Here I missed. It definitely could be moved out from the component in this case